From 9645178e6ff97651f00f9355929c637c4250fa50 Mon Sep 17 00:00:00 2001
From: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
Date: Fri, 12 Aug 2016 08:35:22 +0700
Subject: [PATCH] FDPM user module update version to 1.5.0

Currently, version of FDPM user module is 1.3.0 (source code from driver package of RZ)
While source code of FDPM user module from Yocto 1.10 of R-Car has version 1.5.0
This patch file update version of FDPM user module to 1.5.0 base on Yocto 1.10 of R-Car

Signed-off-by: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
---
 if/Makefile         | 6 +++---
 if/fdpm_api_timer.c | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/if/Makefile b/if/Makefile
index 2295092..436eb27 100755
--- a/if/Makefile
+++ b/if/Makefile
@@ -6,8 +6,8 @@ all:
 	${CC} -fPIC -I$(PWD)/../include -g -c fdpm_api_sub.c -Wall 
 	${CC} -fPIC -I$(PWD)/../include -g -c fdpm_api_fd.c -Wall
 	${CC} -fPIC -I$(PWD)/../include -g -c fdpm_api_timer.c -Wall -lrt
-	${CC} -shared -Wl,-soname=libfdpm.so.1 -o libfdpm.so.1.3.0 fdpm_api.o fdpm_api_sub.o fdpm_api_fd.o fdpm_api_timer.o
-	ln -s ./libfdpm.so.1.3.0 libfdpm.so.1
+	${CC} -shared -Wl,-soname=libfdpm.so.1 -o libfdpm.so.1.5.0 fdpm_api.o fdpm_api_sub.o fdpm_api_fd.o fdpm_api_timer.o
+	ln -s ./libfdpm.so.1.5.0 libfdpm.so.1
 	ln -s ./libfdpm.so.1 libfdpm.so
 clean:
 	rm -f fdpm_api.o
@@ -15,5 +15,5 @@ clean:
 	rm -f fdpm_api_fd.o
 	rm -f fdpm_api_timer.o
 	rm -f libfdpm.so
-	rm -f libfdpm.so.1.3.0
+	rm -f libfdpm.so.1.5.0
 	rm -f libfdpm.so.1
diff --git a/if/fdpm_api_timer.c b/if/fdpm_api_timer.c
index 8103345..f79b6f6 100755
--- a/if/fdpm_api_timer.c
+++ b/if/fdpm_api_timer.c
@@ -43,7 +43,7 @@ int fdpm_timer_open(unsigned char timer_ch, int cb_no, struct fdpm_if_data *idat
 	idata->timer_table[icb_no].se.sigev_notify_attributes = NULL;
 	idata->timer_table[icb_no].se.sigev_signo  = SIGRTMIN + timer_ch + (cb_no%32);
 
-	if (timer_create(CLOCK_REALTIME, &idata->timer_table[icb_no].se, &idata->timer_table[icb_no].timer_id) == -1) {
+	if (timer_create(CLOCK_MONOTONIC, &idata->timer_table[icb_no].se, &idata->timer_table[icb_no].timer_id) == -1) {
 		printf("timer_create failed\n");
 		ret =  -EFAULT;
 		goto exit;
-- 
1.9.1

